pallets/creditcoin/src/helpers/external_address.rs

```diff
@@ -1,14 +1,63 @@
 use crate::{Blockchain, ExternalAddress};
 use base58::FromBase58;
+use core::convert::TryFrom;
+use frame_support::BoundedVec;
+use sp_core::ecdsa::Public;
+use sp_io::hashing::keccak_256;
 use sp_io::hashing::sha2_256;
 
+pub fn external_address_generator(
+	blockchain: &Blockchain,
+	reference: &ExternalAddress,
```

Comment:
what is the intuition behind the name `reference`?

Reply:
it will infer the address protocol from an example.